{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Train ADAGE models on RNAseq data\n",
    "\n",
    "### Georgia Doing 2021\n",
    "\n",
    "This notebook walks through the training of an RNAseq-based ADAGE model as a continuation of analyses from last year (2021_06_eADAGE_adapt/seqADAGE).\n",
    "\n",
    "Since then we have downloaded a new compendium of RNAseq data and aligned it to the PAO1 and PA14 reference genomes using prokarytoic-optimized parameters.\n",
    "(https://github.com/hoganlab-dartmouth/pa-seq-compendia)\n",
    "\n",
    "Section outline:\n",
    "\n",
    "### 1. Setup and Load data\n",
    "    \n",
    "    1.1 libraries and reload for dev\n",
    "    1.2 load ADAGE trained on array data with Jie's original code\n",
    "### 2. Train models with different param options\n",
    "\n",
    "    2.1 USing seed 560, train on different params\n",
    "    2.2 Load models trained with seed 560, on different params, from files\n",
    "### 3. Train replicate models for ensemble\n",
    "    3.1 Use seed 661 with different params\n",
    "    3.2 Use seeds 662-669 to see how models with same params look\n",
    "    3.3 Train models with original ADAGE params\n",
    "    3.4 Train with chosen params, seeds up to 759 for ensemble\n",
    "### 4. Try for deep count autoencoder"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Setup and load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import run_count_autoencoder\n",
    "import run_model\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "from imp import reload\n",
    "import Adage\n",
    "from scipy.stats import hypergeom\n",
    "import csv\n",
    "import TiedWeightsEncoder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for dev purposes, while updating run_model.py need to reload\n",
    "reload(run_count_autoencoder)\n",
    "reload(run_model)\n",
    "reload(Adage)\n",
    "reload(TiedWeightsEncoder)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.1 Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "array_comp = pd.read_csv(open('../data_files/train_set_normalized.csv', \"rb\"),index_col=0)\n",
    "seq_comp_floor = pd.read_csv(open('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log.csv', \"rb\"),index_col=0)\n",
    "seq_comp = pd.read_csv(open('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv', \"rb\"),index_col=0)\n",
    "print(seq_comp.shape)\n",
    "print(seq_comp_floor.shape)\n",
    "print(array_comp.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, (ax1, ax2, ax3) = plt.subplots(1,3,figsize=(16 ,4))\n",
    "sns.histplot(array_comp.to_numpy().flatten(),binwidth=0.01, stat='density',ax=ax1)\n",
    "sns.histplot(seq_comp.to_numpy().flatten(),binwidth=0.01,stat='density',ax=ax2)\n",
    "sns.histplot(seq_comp_floor.to_numpy().flatten(), bins=200,stat='density',ax=ax3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1.2 Load ADAGE model trained using Jie's original code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log.csv',lr=.0001,seed=460,kl1=1e-10,kl2=1e-1, act = 'sigmoid', tied = True, epochs=5, init='glorot_uniform', batch_size=10, v=0)\n",
    "tf_adage = Adage.Adage(m.autoencoder, m.history, m.compendium)\n",
    "tf_weights = np.array(pd.read_csv('../outputs/ADAGE_OG_weights.csv', header = None))\n",
    "tf_cost = pd.read_csv('../outputs/ADAGE_OG_cost_log.csv')\n",
    "tf_adage.weights = tf_weights\n",
    "tf_adage.loss = tf_cost['cost']\n",
    "tf_adage.set_hwg_cutoff(2.5).shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "model_dict = {\n",
    "    \"tf_adage\": tf_adage,\n",
    "    \"k_adage\": m\n",
    "}\n",
    "fig, ax = plt.subplots(1, 2,figsize=(16 ,4))\n",
    "fig.tight_layout(pad=3.0)\n",
    "\n",
    "name = 'tf_adage'\n",
    "model_temp = model_dict[name]\n",
    "ax[0].plot(list(range(0,5)), model_temp.loss[0:5], linewidth=1, markersize=2, color = 'orange')\n",
    "ax[0].set(title = name, xlabel = 'Epochs', ylabel = 'Loss')\n",
    "ax[0].set(title = name)\n",
    "\n",
    "name = 'k_adage'\n",
    "model_temp = model_dict[name]\n",
    "ax[1].plot(list(range(0,5)), model_temp.loss, linewidth=1, markersize=2, color = 'orange')\n",
    "ax[1].plot(list(range(0,5)), model_temp.val_loss, linewidth=1, markersize=2, color = 'red')\n",
    "ax[1].set(title = name, xlabel = 'Epochs', ylabel = 'Loss')\n",
    "ax[1].set(title = name)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Train models to compare paramters (seeds starting at 550)\n",
    "\n",
    "### 2.1 Train models with seed 550 (or skip to 2.1 to load from files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# skip this chunk to load models from files, go to chunk below\n",
    "inits = ['glorot_normal']\n",
    "L1_norm = [1e-10] # \n",
    "L2_norm = [0, 1e-5, 1e-10]\n",
    "act_fun = ['sigmoid', 'tanh','relu']\n",
    "tied = [True, False]\n",
    "lr = [1.5,1.0,0.5,0.1,0.01,0.001,0.0001] \n",
    "\n",
    "model_dict_da = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_dca = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_dca01 = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_arr = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_floor = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "\n",
    "for seed in range(1):\n",
    "    for w in tied:\n",
    "        for i in inits:\n",
    "            #print(i)\n",
    "            for l in L1_norm:\n",
    "                #print(l)\n",
    "                for n in L2_norm:\n",
    "                    for a in act_fun:\n",
    "                       # print(a)\n",
    "                        for t in lr:\n",
    "                            #print(t)\n",
    "                            name = 'ad_' + i + '_' + str(l) + '_' + str(n) + '_' + a + '_tied' + '_' + str(w) + '_' + '_lr' + str(t) \n",
    "                            print(name)\n",
    "                            m_arr = run_model.run_model('../data_files/train_set_normalized.csv',seed=seed+560,lr=t,kl1=l, kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_dca01 = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_dca = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_dca_floor = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_floor.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "\n",
    "\n",
    "                            model_dict_arr[name] = m_arr\n",
    "                            model_dict_da[name] = m_da\n",
    "                            model_dict_dca01[name] = m_dca01\n",
    "                            model_dict_dca[name] = m_dca\n",
    "                            model_dict_floor[name] = m_dca_floor"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.2 Or try to load models from files first, to save time re-training param combos already tested (seed still 550)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "inits = ['glorot_uniform','glorot_normal']\n",
    "L1_norm = [0, 1e-5, 1e-10] # \n",
    "L2_norm = [0, 1e-5, 1e-10]\n",
    "act_fun = ['sigmoid', 'tanh','relu']\n",
    "tied = [True, False]\n",
    "lr = [1.5,1.0,0.5,0.1,0.01,0.001,0.0001] \n",
    "\n",
    "model_dict_da = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_dca = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_dca01 = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_arr = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_floor = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "\n",
    "\n",
    "for w in tied:\n",
    "    for i in inits:\n",
    "        for l in L1_norm:\n",
    "            for n in L2_norm:\n",
    "                for a in act_fun:\n",
    "                    for t in lr:\n",
    "                        seed = str(560)\n",
    "                        name = 'ad_' + i + '_' + str(l) + '_' + str(n) + '_' + a + '_tied' + '_' + str(w) + '_' + '_lr' + str(t) \n",
    "                        #print(name)\n",
    "                    #print(t)\n",
    "                    #m = run_model.run_model('data_files/train_set_normalized.csv',seed=460,kl1=l, act = a, tied = t, epochs=500, init=i)\n",
    "                        prefix = '/data_files/train_set_normalized_seed:' + seed + '_kl1:' + str(l) + '_kl2:0_act:' + a + '_init:' + i + '_ep:50_tied:' + str(w) + '_batch:10_lr:' + str(t)\n",
    "                    #print(prefix)\n",
    "                        try:\n",
    "                            weights_temp = np.array(pd.read_csv('../outputs/weights/' + prefix + '_en_weights_da.csv', header = None))\n",
    "                            loss_temp = np.array(pd.read_csv('../outputs/loss/' + prefix + '_loss_da.csv', header = None))\n",
    "                            val_loss_temp = np.array(pd.read_csv('../outputs/val_loss/' + prefix + '_val_loss_da.csv', header = None))\n",
    "                            adage_temp = Adage.Adage(tf_adage.autoencoder, tf_adage.history, tf_adage.compendium)\n",
    "                            adage_temp.weights = weights_temp\n",
    "                            adage_temp.loss = loss_temp[0]\n",
    "                            adage_temp.val_loss = val_loss_temp[0]\n",
    "                            adage_temp.set_hwg_cutoff(2.5)\n",
    "                        #name = 'ad_' + i + '_' + str(l) + '_' + a + '_tied' + str(t) + seed\n",
    "                    #print(name)\n",
    "                            model_dict_da[name] = adage_temp\n",
    "                        except:\n",
    "                            name = 'ad_' + i + '_' + str(l) + '_' + str(n) + '_' + a + '_tied' + '_' + str(w) + '_' + '_lr' + str(t) \n",
    "                            print(name)\n",
    "                            seed=0\n",
    "                            m_arr = run_model.run_model('../data_files/train_set_normalized.csv',seed=seed+560,lr=t,kl1=l, kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_dca01 = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_dca = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_dca_floor = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_floor.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "\n",
    "\n",
    "                            model_dict_arr[name] = m_arr\n",
    "                            model_dict_da[name] = m_da\n",
    "                            model_dict_dca01[name] = m_dca01\n",
    "                            model_dict_dca[name] = m_dca\n",
    "                            model_dict_floor[name] = m_dca_floor"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Setting the seed to 660 to train a new set of models\n",
    "\n",
    "### 3.1 First, try different params again with seed 660 to make sure all looks same as seed 560"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "inits = ['glorot_uniform']\n",
    "L1_norm = [0, 1e-5, 1e-10,1e-15]\n",
    "L2_norm = [0]\n",
    "act_fun = ['sigmoid', 'tanh','relu']\n",
    "tied = [True]\n",
    "lr = [1.0,0.5,0.1,0.01] #1.5,\n",
    "\n",
    "for seed in range(1):\n",
    "    for w in tied:\n",
    "        for i in inits:\n",
    "            #print(i)\n",
    "            for l in L1_norm:\n",
    "                #print(l)\n",
    "                for n in L2_norm:\n",
    "                    for a in act_fun:\n",
    "                       # print(a)\n",
    "                        for t in lr:\n",
    "                            #print(t)\n",
    "                            name = 'ad_' + i + '_' + str(l) + '_' + str(n) + '_' + a + '_tied' + '_' + str(w) + '_' + '_lr' + str(t) \n",
    "                            print(name)\n",
    "                            m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+660,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=250, init=i, batch_size = 10,v=0)\n",
    "                                "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.2 Now, train 8 new models with seeds 662 - 670\n",
    "(starting with 8 to look at enrichments quick before scaling up even more)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "inits = ['glorot_uniform']\n",
    "L1_norm = [0, 1e-5, 1e-10,1e-15]\n",
    "L2_norm = [0]\n",
    "act_fun = ['sigmoid', 'tanh','relu']\n",
    "tied = [True]\n",
    "lr = [1.0,0.5,0.1,0.01] #1.5,\n",
    "\n",
    "for seed in range(2,10):\n",
    "    for w in tied:\n",
    "        for i in inits:\n",
    "            #print(i)\n",
    "            for l in L1_norm:\n",
    "                #print(l)\n",
    "                for n in L2_norm:\n",
    "                    for a in act_fun:\n",
    "                       # print(a)\n",
    "                        for t in lr:\n",
    "                            #print(t)\n",
    "                            name = 'ad_' + i + '_' + str(l) + '_' + str(n) + '_' + a + '_tied' + '_' + str(w) + '_' + '_lr' + str(t) \n",
    "                            print(name)\n",
    "                            #m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01_bygene.csv',seed=seed+660,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=250, init=i, batch_size = 10,v=0)\n",
    "                            m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01_byall.csv',seed=seed+660,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=250, init=i, batch_size = 10,v=0)\n",
    "                            #m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+660,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=250, init=i, batch_size = 10,v=0)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.3 Train some models with original params and seed 660 \n",
    "Note this data is normalized by gene, activation is sigmoid - matched original parameters of ADAGE\n",
    "\n",
    "* hidden features: 300\n",
    "* epochs: 250\n",
    "* curruption: 0.01\n",
    "* batchs zie: 10\n",
    "* validatoin split: 0.1\n",
    "* activation: sigmoid\n",
    "* objective: binary cross-entropy\n",
    "* learning rate: 0.1\n",
    "* L1: 1e-10\n",
    "* tied: True\n",
    "* init: uniform"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for seed in range(1):\n",
    "    #m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01_bygene.csv',seed=seed+660,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=250, init=i, batch_size = 10,v=0)\n",
    "    run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01_bygene.csv',seed=seed+660,lr=1.0,kl1=1e-10,kl2=0, act = 'sigmoid', tied = True, epochs=250, init='glorot_uniform', batch_size = 10,v=0)\n",
    "    #m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+660,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=250, init=i, batch_size = 10,v=0)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.4 Now, finally,  train the models to get a collection of 100 models with seeds 660 - 759\n",
    "(this loop previously ran range(0,75)\n",
    "\n",
    "* hidden features: 300\n",
    "* epochs: 250\n",
    "* curruption: 0.01\n",
    "* batchs zie: 10\n",
    "* validatoin split: 0.1\n",
    "* activation: relu\n",
    "* objective: binary cross-entropy\n",
    "* learning rate: 0.1\n",
    "* L1: 1e-5\n",
    "* tied: True\n",
    "* init: uniform"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for seed in range(76,100):\n",
    "    #m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01_bygene.csv',seed=seed+660,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=250, init=i, batch_size = 10,v=0)\n",
    "    run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01_byall.csv',seed=seed+660,lr=0.1,kl1=1e-5,kl2=0, act = 'relu', tied = True, epochs=250, init='glorot_uniform', batch_size = 10,v=0)\n",
    "    #m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+660,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=250, init=i, batch_size = 10,v=0)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Try DCA (count autoencoder)\n",
    "seed back to 560"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "inits = ['glorot_uniform']\n",
    "L1_norm = [0, 1e-10,1e-15] #1e-5,\n",
    "L2_norm = [1e-5,1e-10]\n",
    "act_fun = [ 'sigmoid','tanh','relu']#\n",
    "tied = [True]\n",
    "lr = [1.5,1.0,0.5,0.1,0.01,0.001] #\n",
    "\n",
    "for seed in range(1):\n",
    "    for w in tied:\n",
    "        for i in inits:\n",
    "            #print(i)\n",
    "            for l in L1_norm:\n",
    "                #print(l)\n",
    "                for n in L2_norm:\n",
    "                    for a in act_fun:\n",
    "                       # print(a)\n",
    "                        for t in lr:\n",
    "                            #print(t)\n",
    "                            name = 'ad_' + i + '_' + str(l) + '_' + str(n) + '_' + a + '_tied' + '_' + str(w) + '_' + '_lr' + str(t) \n",
    "                            print(name)\n",
    "                            m_da = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01_byall.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_da_bg = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01_bygene.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            #m_da_bs = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_TPM_log_01_bysamp.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "inits = ['glorot_normal','glorot_uniform']\n",
    "L1_norm = [0, 1e-5, 1e-10] # \n",
    "L2_norm = [0, 1e-5, 1e-10]\n",
    "act_fun = ['sigmoid', 'tanh','relu']\n",
    "tied = [True, False]\n",
    "lr = [1.5,1.0,0.5,0.1,0.01,0.001,0.0001] \n",
    "\n",
    "model_dict_da = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_dca = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_dca01 = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_arr = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "model_dict_floor = {\n",
    "    \"tf_adage\": tf_adage\n",
    "}\n",
    "\n",
    "for seed in range(1):\n",
    "    for w in tied:\n",
    "        for i in inits:\n",
    "            #print(i)\n",
    "            for l in L1_norm:\n",
    "                #print(l)\n",
    "                for n in L2_norm:\n",
    "                    for a in act_fun:\n",
    "                       # print(a)\n",
    "                        for t in lr:\n",
    "                            #print(t)\n",
    "                            name = 'ad_' + i + '_' + str(l) + '_' + str(n) + '_' + a + '_tied' + '_' + str(w) + '_' + '_lr' + str(t) \n",
    "                            print(name)\n",
    "                            m_arr = run_model.run_model('../data_files/train_set_normalized.csv',seed=seed+560,lr=t,kl1=l, kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            m_da = run_model.run_model('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            #m_dca01 = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log_01.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            #m_dca = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_log.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "                            #m_dca_floor = run_count_autoencoder.run_count_autoencoder('../data_files/pao1_aligned_rnaseq_compendium_zp2_MRnorm_floor.csv',seed=seed+560,lr=t,kl1=l,kl2=n, act = a, tied = w, epochs=50, init=i, batch_size = 10,v=0)\n",
    "\n",
    "\n",
    "                            model_dict_arr[name] = m_arr\n",
    "                            model_dict_da[name] = m_da\n",
    "                            #model_dict_dca01[name] = m_dca01\n",
    "                            #model_dict_dca[name] = m_dca\n",
    "                            #model_dict_floor[name] = m_dca_floor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(model_dict_da.keys())\n",
    "print(model_dict_dca.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "yd = len(inits) * len(L1_norm)\n",
    "xd = len(act_fun) * len(lr)\n",
    "fig, ax = plt.subplots(xd, yd,figsize=(xd*4 ,yd *4))\n",
    "fig.tight_layout(pad=3.0)\n",
    "\n",
    "xi = 0\n",
    "yi = 0\n",
    "for l in L1_norm:\n",
    "    for i in inits:\n",
    "        xi = 0\n",
    "        for a in act_fun:\n",
    "            for t in lr:\n",
    "                name = 'ad_' + i + '_' + str(l) + '_' + a + '_tied' + str(t) \n",
    "                #print(name)\n",
    "                model_temp = model_dict_da[name]\n",
    "                ax[xi,yi].plot(list(range(0,50)), model_temp.loss, linewidth=1, markersize=2, color = 'orange')\n",
    "                ax[xi,yi].set(title = name, xlabel = 'Epochs', ylabel = 'Loss')\n",
    "                xi = xi+1\n",
    "        yi=yi+1\n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "yd = len(inits) * len(L1_norm)\n",
    "xd = len(act_fun) * len(lr)\n",
    "fig, ax = plt.subplots(xd, yd,figsize=(xd*4 ,yd *4))\n",
    "fig.tight_layout(pad=3.0)\n",
    "\n",
    "xi = 0\n",
    "yi = 0\n",
    "for l in L1_norm:\n",
    "    for i in inits:\n",
    "        xi = 0\n",
    "        for a in act_fun:\n",
    "            for t in lr:\n",
    "                name = 'ad_' + i + '_' + str(l) + '_' + a + '_tied' + str(t) \n",
    "                #print(name)\n",
    "                model_temp = model_dict_dca[name]\n",
    "                ax[xi,yi].plot(list(range(0,50)), model_temp.loss, linewidth=1, markersize=2, color = 'orange')\n",
    "                ax[xi,yi].set(title = name, xlabel = 'Epochs', ylabel = 'Loss')\n",
    "                xi = xi+1\n",
    "        yi=yi+1\n",
    "\n",
    "\n",
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
